src/query/functions/src/scalars/string.rs

```diff
@@ -841,6 +854,190 @@ pub(crate) mod soundex {
+                DataWrapper::Bitwise(v1) => *v1 |= 1 << idx,
+            }
+        }
+    }
+
+    /// Calculates the Jaro-Winkler distance of two strings.
+    ///
+    /// The return value is between 0.0 and 1.0, where 1.0 means the strings are equal.
+    pub fn jaro_winkler(left_: &str, right_: &str) -> f64 {
+        let llen = left_.len();
+        let rlen = right_.len();
+
+        let (left, right, s1_len, s2_len) = if llen < rlen {
+            (right_, left_, rlen, llen)
+        } else {
+            (left_, right_, llen, rlen)
+        };
+
+        match (s1_len, s2_len) {
+            (0, 0) => return 1.0,
+            (0, _) | (_, 0) => return 0.0,
+            (_, _) => (),
+        }
+
+        if left == right {
+            return 1.0;
+        }
+
+        let range = matching_distance(s1_len, s2_len);
+        let mut s1m = DataWrapper::build(s1_len);
+        let mut s2m = DataWrapper::build(s2_len);
+        let mut matching: f64 = 0.0;
+        let mut transpositions: f64 = 0.0;
+        let left_as_bytes = left.as_bytes();
+        let right_as_bytes = right.as_bytes();
+
+        for i in 0..s2_len {
+            let mut j = (i as isize - range as isize).max(0) as usize;
+            let l = (i + range + 1).min(s1_len);
+            while j < l {
+                if right_as_bytes[i] == left_as_bytes[j] && !s1m.get(j) {
+                    s1m.set_true(j);
+                    s2m.set_true(i);
+                    matching += 1.0;
+                    break;
+                }
+
+                j += 1;
+            }
+        }
+
+        if matching == 0.0 {
+            return 0.0;
+        }
+
+        let mut l = 0;
+
+        for i in 0..s2_len - 1 {
+            if s2m.get(i) {
+                let mut j = l;
+
+                while j < s1_len {
+                    if s1m.get(j) {
+                        l = j + 1;
+                        break;
+                    }
+
+                    j += 1;
+                }
+
+                if right_as_bytes[i] != left_as_bytes[j] {
+                    transpositions += 1.0;
+                }
+            }
+        }
+        transpositions = (transpositions / 2.0).ceil();
+
+        let jaro = (matching / (s1_len as f64)
+            + matching / (s2_len as f64)
+            + (matching - transpositions) / matching)
+            / 3.0;
+
+        let prefix_length = left_as_bytes
+            .iter()
+            .zip(right_as_bytes)
+            .take(4)
+            .take_while(|(l, r)| l == r)
+            .count() as f64;
+
+        jaro + prefix_length * 0.1 * (1.0 - jaro)
+    }
+
+    fn matching_distance(s1_len: usize, s2_len: usize) -> usize {
+        let max = s1_len.max(s2_len) as f32;
+        ((max / 2.0).floor() - 1.0) as usize
+    }
+
+#[cfg(test)]
+    mod tests {
+        use super::*;
+
+        #[test]
+        fn different_is_zero() {
+            assert_eq!(jaro_winkler("foo", "bar"), 0.0);
```

Comment:
Please impl the ut in https://github.com/datafuselabs/databend/blob/629ee7bab86bf74538814e794122e1fedbbeb288/src/query/functions/tests/it/scalars/other.rs